h/h_api/bulk_api/model/data_body.py

```diff
@@ -5,55 +5,42 @@
 from h.h_api.schema import Schema
 
 
-class UpsertUser(JSONAPIData):
-    """The data to upsert a user."""
-
-    validator = Schema.get_validator("bulk_api/command/upsert_user.json")
+class UpsertBody(JSONAPIData):
+    data_type = None
+    query_fields = []
 
     @classmethod
     def create(cls, attributes, id_reference):
-        """
-        Create an upsert user body.
-
-        :param _id: User id
-        :param attributes: User attributes
-        :return:
-        """
-
-        authority = attributes.pop("authority", None)
-        username = attributes.pop("username", None)
+        query = {field: attributes.pop(field, None) for field in cls.query_fields}
 
         return super().create(
-            DataType.USER,
+            data_type=cls.data_type,
             attributes=attributes,
-            meta={
-                "query": {"authority": authority, "username": username},
-                "$anchor": id_reference,
-            },
+            meta={"query": query},
+            id_reference=id_reference,
         )
 
+    @property
+    def query(self):
+        """The query used to select which item to update."""
```

Comment:
The `UpsertUser` and `UpsertGroup` can now be unified as they are basically identical